addons/wowl/static/src/components/loading_overlay/loading_overlay.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, useState } from "@odoo/owl";
+import type { OdooEnv } from "../../types";
+import { LoadingSpinner } from "./loading_spinner";
+
+export class LoadingOverlay extends Component<{}, OdooEnv> {
+  static template = "wowl.LoadingOverlay";
+  static components = { LoadingSpinner };
+  state = useState({
+    blockUI: false,
+  });
+
+  constructor() {
+    super(...arguments);
+    this.env.bus.on("LOADING_OVERLAY:BLOCKUI", this, () => {
+      this._blockUI();
```

Comment:
This is interesting. I was thinking about more or less the same thing last week, specifically on the blockUI stuff.
I guess I would have done essentially what you did.
except I would have put a service that would have triggered the event BLOCKUI / UNBLOCKUI
Blocking the ui is not specific to the loading overlay, instead, a flow may want to block the ui, then trigger some async stuff, and when the action returns, unblock the ui (when printing a report, the ui is blocked right away)